recipes/log4cplus/all/conanfile.py

```diff
@@ -1,110 +1,106 @@
-        self._cmake.definitions["UNICODE"] = self.options.unicode
-        self._cmake.definitions["LOG4CPLUS_BUILD_TESTING"] = False
-        self._cmake.definitions["WITH_UNIT_TESTS"] = False
-        self._cmake.definitions["LOG4CPLUS_ENABLE_DECORATED_LIBRARY_NAME"] = self.options.decorated_name
-        self._cmake.definitions["LOG4CPLUS_QT4"] = False
-        self._cmake.definitions["LOG4CPLUS_QT5"] = False
-        self._cmake.definitions["LOG4CPLUS_SINGLE_THREADED"] = self.options.single_threaded
-        self._cmake.definitions["LOG4CPLUS_BUILD_LOGGINGSERVER"] = self.options.build_logging_server
-        self._cmake.definitions["WITH_ICONV"] = self.options.with_iconv
-        self._cmake.definitions["LOG4CPLUS_WORKING_LOCALE"] = self.options.working_locale
-        self._cmake.definitions["LOG4CPLUS_WORKING_C_LOCALE"] = self.options.working_c_locale
-        self._cmake.configure(build_dir=self._build_subfolder)
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["LOG4CPLUS_ENABLE_THREAD_POOL"] = self.options.thread_pool
+        tc.variables["UNICODE"] = self.options.unicode
+        tc.variables["LOG4CPLUS_BUILD_TESTING"] = False
+        tc.variables["WITH_UNIT_TESTS"] = False
+        tc.variables["LOG4CPLUS_ENABLE_DECORATED_LIBRARY_NAME"] = self.options.decorated_name
+        tc.variables["LOG4CPLUS_QT4"] = False
+        tc.variables["LOG4CPLUS_QT5"] = False
+        tc.variables["LOG4CPLUS_SINGLE_THREADED"] = self.options.single_threaded
+        tc.variables["LOG4CPLUS_BUILD_LOGGINGSERVER"] = self.options.build_logging_server
+        tc.variables["WITH_ICONV"] = self.options.with_iconv
+        tc.variables["LOG4CPLUS_WORKING_LOCALE"] = self.options.working_locale
+        tc.variables["LOG4CPLUS_WORKING_C_LOCALE"] = self.options.working_c_locale
+        tc.generate()
 
-    def _patch_sources(self):
-        # don't force PIC
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                              "set (CMAKE_POSITION_INDEPENDENT_CODE ON)", "")
+        dpes = CMakeDeps(self)
+        dpes.generate()
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.libs = collect_libs(self)
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "log4cplus")
        self.cpp_info.set_property("cmake_target_name", "log4cplus::log4cplus")
        self.cpp_info.libs = collect_libs(self)
```

There is config file upstream, so these properties can be enforced.